### DISCUSSION TITLE
Nodes versus vertices

### DISCUSSION THREAD

**Discussion by erlebach** [JuliaGraphs/MultilayerGraphs.jl#95] [General]
What is the difference between nodes and multivertices in a MULTIGRAPHS? I noticed that the multivertex structure contains a node. Is each  later comprised of multiple multivertices, while different multivertices can correspond to the same node? If so, why not replace the Node by the Node Id in the multivertex structure? Thanks!

**Comment by InterdisciplinaryPhysicsTeam** [comment]
Hi @erlebach,

Thanks for the feedback!

To better explain the difference between a `Node` and a `MultilayerVertex`, we just added one more sentence to the documentation:

> A multilayer graph is composed of layers, i.e. graphs whose vertices represent the same set of nodes (not all nodes need to be represented in every layer), and interlayers, i.e. the bipartite graphs that connect vertices in two different layers. Vertices in a multilayer graph are represented using the `MultilayerVertex` struct, while nodes are represented using the `Node` struct.

That you may find [here](https://juliagraphs.org/MultilayerGraphs.jl/dev/#:~:text=A%20multilayer%20graph%20is%20composed,represented%20using%20the%20Node%20struct.) for now. Will be merged into the stable documentation after the next release.

Put in other words, multilayer graphs are a collection of layers (and interlayers, but they don't matter to us now). Each vertex in each layer (i.e. each `MultilayerVertex`) represents one `Node` from a single set of nodes which is a property of the multilayer graph. Thus, a `MultilyerVertex` essentially carries two pieces of information: the `Node` it represents, and the name of the layer it represents the `Node` into (i.e. the name of the layer it belongs to). A `MultilayerVertex` represents a `Node` in a `Layer`, and thus a `Node` may be represented in multiple `Layer`s.

So `MultilyerVertex`s are the closest to being the multilayer equivalent to standard integer-labeled vertices that you may find in e.g. `SimpleGraph`s from `Graph.jl` (infact they are internally mapped to one such representation). 

There is an ongoing discussion (#68) about removing `Node`s in favor of integer labels. This would have pros and cons. Anyway, we currently feel like substituting them with strings (i.e. the `Node`'s id you are proposing) will retain all the cons of not having an integer representation, while disallowing us from modifying the `Node`s' behavior in the future without introducing breaking changes (since we would no longer have a custom struct to work with).
